std/traits.d

```diff
@@ -5604,12 +5627,41 @@ enum bool isSomeString(T) = is(StringTypeOf!T) && !isAggregateType!T && !isStati
     }
 }
 
-/**
- * Detect whether type $(D T) is a narrow string.
- *
- * All arrays that use char, wchar, and their qualified versions are narrow
- * strings. (Those include string and wstring).
- */
+/++
+    Detect whether type `T` is a narrow string.
+
+    Any array type whose element type is `char` or `wchar` - regardless of
+    qualifiers - is a narrow string (that includes `string` and `wstring`). It
+    specifically does $(I not) include arrays of `dchar`.
+
+    The reason that the concept of a narrow string is important is that the
+    range primitives (`front`, `popFront`, `back`, etc.) auto-decode narrow
+    strings so that they're ranges of `dchar` rather than their actual element
+    type with the idea that it avoids processing pieces of code points
+    separately (though it does not avoid processing pieces of graphemes
+    separately). Also $(REF hasSlicing, std,range,primitives) and
+    $(REF hasLength, std,range,primitives),
+    $(REF isRandomAccessRange, std,range,primitives) are considered to be
+    `false` for narrow strings (in order to avoid splitting up a code point).
+    isNarrowString makes it easy to specialize code for narrow strings in order
+    to avoid unnecessary auto-decoding or to have more control over the
+    decoding. Alternatively, Phobos contains helper functions for wrapping
+    ranges of characters in other ranges which turn them into ranges of code
+    units, code points, or graphemes.
+
+    See_Also: $(REF ElementEncodingType, std,range,primitives),
+        $(REF byCodeUnit, std,utf), $(REF byUTF, std,utf),
+        $(REF byGrapheme, std,uni)
+
+    $(RED Warning:
+        As with $(LREF isSomeString), isNarrowString is true for enums whose
+        base type is a string type (see the warning on $(LREF isSomeString)).
+        This is less of an issue with isNarrowString than $(LREF isSomeString),
```

Comment:
This last part seems a bit unnecessary -- it sounds like a comment on the state of things rather than a documentation per se of `isNarrowString`. I'd shorten it to something along the lines of "This may cause issues if explicit conversion to the base type is necessary for the code to work."